models/CouponModel.php

```diff
@@ -924,6 +931,10 @@ public function is_coupon_applicable( object $coupon, int $object_id, string $or
 			}
 		}
 
+		if ( ! $is_applicable ) {
+			$coupon_apply_error_msg = $this->get_coupon_failed_error_msg( 'specific_applicable', str_replace( '_', ' ', $applies_to ) );
+		}
+
 		return apply_filters( 'tutor_coupon_is_applicable', $is_applicable, $coupon, $object_id, $applications, $is_membership_plan );
```

Comment:
Here also. Need to set error after the filter result.

Reply:
Setting the error message before or after the filter makes no difference!